hazelcast/src/test/java/com/hazelcast/map/EntryProcessorExpiryTimeTest.java

```diff
@@ -0,0 +1,305 @@
+                new SetValueWithoutChangingExpiryTimeOffloadable<>(2), false);
+    }
+
+    @Test
+    public void executeOnKeys_sets_expiry_time_when_creating_new_entries_with_offloadable_EP() {
+        test(new HashSet<>(Arrays.asList(1, 2, 3, 4, 5, 6, 7)), false,
+                new SetValueWithoutChangingExpiryTimeOffloadable<>(2), false);
+    }
+
+    @Test
+    public void executeOnKey_does_not_change_expiry_time_when_updating_entry_with_offloadable_EP() {
+        test(Collections.singleton(1), true,
+                new SetValueWithoutChangingExpiryTimeOffloadable<>(2), false);
+    }
+
+    @Test
+    public void executeOnKeys_does_not_change_expiry_time_when_updating_entries_with_offloadable_EP() {
+        test(new HashSet<>(Arrays.asList(1, 2, 3, 4, 5, 6, 7)), true,
+                new SetValueWithoutChangingExpiryTimeOffloadable<>(2), false);
+    }
+
+    @Test
+    public void executeOnKey_does_not_change_expiry_time_when_updating_entries_with_offloadable_EP() {
+        test(new HashSet<>(Arrays.asList(1, 2, 3, 4, 5, 6, 7)), true,
+                new SetValueWithoutChangingExpiryTimeOffloadable<>(2), false);
+    }
+
+    @Test
+    public void first_setValueWithoutChangingExpiryTime_then_setValue_shifts_expiry_time() {
+        test(new HashSet<>(Arrays.asList(1, 2, 3, 4, 5, 6, 7)), true,
+                new First_SetValueWithoutChangingExpiryTime_Then_SetValue<>(2), true);
+    }
+
+    @Test
+    public void first_setValue_then_setValueWithoutChangingExpiryTime_does_not_shift_expiry_time() {
+        test(new HashSet<>(Arrays.asList(1, 2, 3, 4, 5, 6, 7)), true,
+                new First_SetValue_Then_SetValueWithoutChangingExpiryTime(2), false);
+    }
+
+    private void test(Set<Integer> keySet, boolean update,
+                      EntryProcessor entryProcessor, boolean expectShiftExpiry) {
+        HazelcastInstance[] hazelcastInstances = createHazelcastInstances(getConfig(), 2);
+        HazelcastInstance instance1 = hazelcastInstances[0];
+        HazelcastInstance instance2 = hazelcastInstances[1];
+
+        String mapName = "test-map";
+        MapBackupAccessor mapBackupAccessor = (MapBackupAccessor) TestBackupUtils
+                .newMapAccessor(hazelcastInstances, mapName, 1);
+
+        IMap<Integer, Integer> instance1Map = instance1.getMap(mapName);
+        Map<Integer, EntryView> cacheEntryViewPerKey1 = null;
+
+        // when update is true, first create entries
+        if (update) {
+            for (Integer key : keySet) {
+                instance1Map.set(key, 1, 100, TimeUnit.SECONDS);
+            }
+            cacheEntryViewPerKey1 = cacheEntryViewPerKey(instance1Map, keySet);
+
+            sleepAtLeastSeconds(1);
+        }
+
+        if (keySet.size() == 1) {
+            Integer key = keySet.iterator().next();
+            instance1Map.executeOnKey(key, entryProcessor);
+        } else {
+            instance1Map.executeOnKeys(keySet, entryProcessor);
+        }
```

Comment:
Maybe we can have `instance1Map.executeOnEntries(EntryProcessor entryProcessor)` and `instance1Map.executeOnEntries(EntryProcessor entryProcessor, Predicate predicate)` tests to test every public API.

For example currently changes in `PartitionWideEntryOperation` and `PartitionWideEntryBackupOperation` are untested.

Reply:
done

Reply:
Thank you